docs/source/kedro_datasets.rst

```diff
@@ -0,0 +1,52 @@
+kedro_datasets
+==============
+
+.. rubric:: Description
+
+.. automodule:: kedro_datasets
+
+.. rubric:: Classes
+
+.. autosummary::
+   :toctree:
+   :template: autosummary/class.rst
+
+   kedro_datasets.api.APIDataSet
```

Comment:
@merelcht here's the magic. `autodoc` and `sphinx.ext.autosummary` only need to be able to _import_ the objects, they don't even have to be downloaded physically!

Reply:
So you need to have `kedro-datasets` installed? And what if you have an older version, would it then build old docs?

Reply:
That's correct, it will load whatever version is installed. Since the `docs` extras-requires is getting installed, `kedro-datasets` and all its dependencies were already there:

https://github.com/kedro-org/kedro/blob/a95ce7aa4dee49b6c17aeb082c8b88072fa47cf4/Makefile#L36-L37

https://github.com/kedro-org/kedro/blob/a95ce7aa4dee49b6c17aeb082c8b88072fa47cf4/setup.py#L109

Reply:
Alright, so I guess in that case we need to make sure to bump the pin here whenever we do a `kedro-datasets` release to make sure docs are the most recent.

Reply:
I've added checking to bump the `kedro-datasets` doc requirements as a step in the release notes.